modules/oadp-checking-api-group-versions.adoc

```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/advanced-topics.adoc
+
+
+:_content-type: PROCEDURE
+[id="oadp-checking-api-group-versions_{context}"]
+= Checking Kubernetes API group versions
+
+A source cluster might have a resource with multiple Kubernetes API versions registered to it. One of those versions is the preferred version. If you use Velero to back up and restore such a source cluster, Velero backs up only the version of that resource that uses the preferred version of its Kubernetes API.
```

Comment:
So in k8s there might be multiple versions of an API available. So the causality is backwards here, I think. Instead:

> A source cluster might offer multiple versions of an API, where one of these versions is the preferred API version. For example, a source cluster with an API named `Example` might be available in the `example.com/v1` and `example.com/v1beta2` API groups.

Something like that, at any rate.

The key is that only API objects created against these preferred APIs are backed up. For more particulars, there is this: https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/

But we're talking about these:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
spec:
...
```